test/gtest/uct/ib/test_ud.cc

```diff
@@ -918,6 +918,74 @@ UCS_TEST_SKIP_COND_P(test_ud, ctls_loss,
 }
 #endif
 
+#if UCT_UD_EP_DEBUG_HOOKS
+/* Stale ACK PSN to inject - simulates ACK from before endpoint reset */
+static volatile uct_ud_psn_t stale_ack_psn_to_inject = 0;
```

Comment:
Minor. Maybe a better design approach would be to have `stale_ack_psn_to_inject` and `inject_stale_ack_psn` as class members, rather than static?

Can inherit and specify a sub-class.